samples/ask-akka-agent/src/main/java/akka/ask/agent/application/StructuredAskAkkaAgent.java

```diff
@@ -0,0 +1,36 @@
+package akka.ask.agent.application;
+
+import akka.javasdk.agent.Agent;
+import akka.javasdk.agent.ModelProvider;
+import akka.javasdk.annotations.AgentDescription;
+import akka.javasdk.annotations.ComponentId;
+
+@ComponentId("structured-ask-akka-agent")
+@AgentDescription(name = "Akka Design Specialist", description = "Expert in Akka design questions")
+public class StructuredAskAkkaAgent extends Agent {
+  public record Response(String summary, String akkaComponent) {}
+
+  private final String sysMessage = """
+      You are a specialized agent that helps with Akka design question.
+      
+      IMPORTANT: You must respond with a valid JSON object that follows this structure:
+        {
+          "summary": "Brief overview of the design proposal",
+          "akkaComponent": "The name of the Akka component to use"
+        }
+      
+      Do not include any explanations or text outside of the JSON structure.
+      If you are unsure, say the following in the summary field:
+      Sorry, I don't know how to help with that.
+      """; // <1>
+
+  public Effect<Response> ask(String question) {
+    return effects()
+        .modelProvider(ModelProvider.openAi()
```

Comment:
I wonder how we could mock that in tests. We could inject the provider via the constructor and then pass `TestModelProvider` or sth. Anyway, it's good to remember about the mocking aspect.

Reply:
Yes, either we have some specialized testkit or use DI as you say. We should probably have some kind of TestModelProvider. In the end it should be possible for us to implement a langchain4j `ChatModel` for testing purposes.